src/integrationTest/java/uk/gov/hmcts/reform/cpo/wiremock/extension/DynamicS2sDetailsResponseTransformer.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.reform.cpo.wiremock.extension;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.github.tomakehurst.wiremock.extension.Parameters;
+import com.github.tomakehurst.wiremock.http.Request;
+import com.github.tomakehurst.wiremock.http.Response;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Date;
+
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+import static uk.gov.hmcts.reform.cpo.security.SecurityUtils.BEARER;
+
+/*
+ * Replaces response body with service name read from supplied token before returning the response
+ */
+public class DynamicS2sDetailsResponseTransformer extends AbstractDynamicResponseTransformer {
```

Comment:
Does it require unit test ?

Reply:
Unchanged: as this is a utitly class to support the wiremocks used in the integration test.  The tests defined in `BaseMvcAuthChecks` verify both happy and negative endpoint responses using authentication agasint the responses from this stub.